PennMobile/GSR-Booking/Controllers/GSRGroupNewIntialController.swift

```diff
@@ -0,0 +1,284 @@
+        createButton.heightAnchor.constraint(equalToConstant: 50).isActive = true
+        createButton.leftAnchor.constraint(equalTo: view.leftAnchor, constant: 15).isActive = true
+        createButton.rightAnchor.constraint(equalTo: view.rightAnchor, constant: -15).isActive = true
+        createButton.translatesAutoresizingMaskIntoConstraints = false
+        
+        //button unclickable until group name changed and not empty
+        createButton.isUserInteractionEnabled = false;
+        nameChanged = false;
+    }
+
+    func prepareColorLabel() {
+        colorLabel = UILabel()
+        colorLabel.text = "Pick a color:"
+        colorLabel.font = UIFont.systemFont(ofSize: 17)
+        colorLabel.textColor = UIColor.init(red: 153, green: 153, blue: 153)
+        colorLabel.textAlignment = .center
+
+        view.addSubview(colorLabel)
+        colorLabel.leftAnchor.constraint(equalTo: view.leftAnchor, constant: 0).isActive = true
+        colorLabel.rightAnchor.constraint(equalTo: view.rightAnchor, constant: 0).isActive = true
+        colorLabel.topAnchor.constraint(equalTo: barView.bottomAnchor, constant: 35).isActive = true
+        colorLabel.translatesAutoresizingMaskIntoConstraints = false
+    }
+
+    func prepareColorCollection() {
+        let layout = UICollectionViewFlowLayout()
+        layout.scrollDirection = .horizontal
+        layout.minimumLineSpacing = 0
+        layout.sectionInset = UIEdgeInsets(top: 0, left: 0, bottom: 0, right: 0)
+        colorCollectionView = UICollectionView(frame: .zero, collectionViewLayout: layout)
+        colorCollectionView.register(GSRColorCell.self, forCellWithReuseIdentifier: GSRColorCell.identifier)
+        colorCollectionView.backgroundColor = .clear
+        colorCollectionView.showsHorizontalScrollIndicator = false
+        colorCollectionView.delegate = self
+        colorCollectionView.dataSource = self
+        colorCollectionView.allowsSelection = true;
+        colorCollectionView.allowsMultipleSelection = false
+        colorCollectionView.selectItem(at: IndexPath(item: 0, section: 0), animated: false, scrollPosition: UICollectionView.ScrollPosition.centeredHorizontally)
+        view.addSubview(colorCollectionView)
+
+        colorCollectionView.topAnchor.constraint(equalTo: colorLabel.bottomAnchor, constant: 20).isActive = true
+        colorCollectionView.leftAnchor.constraint(equalTo: view.leftAnchor, constant: 30).isActive = true
+        colorCollectionView.rightAnchor.constraint(equalTo: view.rightAnchor, constant: -20).isActive = true
+        colorCollectionView.heightAnchor.constraint(equalToConstant: 40).isActive = true
+        colorCollectionView.translatesAutoresizingMaskIntoConstraints = false
+    }
+
+
+    @objc func createGroupBtnAction(sender:UIButton!) {
+        GSRGroupNetworkManager.instance.createGroup(name: nameField.text!, color: colorLabel.text!) { (success, errorMsg) in
```

Comment:
Let's try to avoid bang operators unless it's obviously guaranteed to never be null. Will this crash if the user presses create group without choosing the name or color?

Reply:
Not my code, but currently the button gets disabled until the user selects a name & color so that they will not be null. I can still fix this in a commit though.